sql/src/main/java/io/crate/operation/reference/sys/node/DiscoveryNodeContextFieldResolver.java

```diff
@@ -0,0 +1,251 @@
+                    }
+                }
+            })
+            .put(SysNodesTableInfo.Columns.REST_URL, new Consumer<DiscoveryNodeContext>() {
+                @Override
+                public void accept(DiscoveryNodeContext context) {
+                    if (clusterService.localNode() != null) {
+                        String url = clusterService.localNode().attributes().get("http_address");
+                        context.restUrl(BytesRefs.toBytesRef(url));
+                    }
+                }
+            })
+            .put(SysNodesTableInfo.Columns.PORT, new Consumer<DiscoveryNodeContext>() {
+                @Override
+                public void accept(DiscoveryNodeContext context) {
+                    Integer http = null;
+                    Integer transport;
+
+                    if (nodeService.info().getHttp() != null) {
+                        http = portFromAddress(nodeService.info().getHttp().address().publishAddress());
+                    }
+                    try {
+                        transport = portFromAddress(nodeService.stats().getNode().address());
+                    } catch (IOException e) {
+                        throw new ElasticsearchException("unable to get node transport statistics", e);
+                    }
+                    HashMap<String, Integer> port = new HashMap<>();
+                    port.put("http", http);
+                    port.put("transport", transport);
+                    context.port(port);
+                }
+            })
+            .put(SysNodesTableInfo.Columns.LOAD, new Consumer<DiscoveryNodeContext>() {
+                @Override
+                public void accept(DiscoveryNodeContext context) {
+                    context.extendedOsStats(extendedNodeInfo.osStats());
+                }
+            })
+            .put(SysNodesTableInfo.Columns.MEM, new Consumer<DiscoveryNodeContext>() {
+                @Override
+                public void accept(DiscoveryNodeContext context) {
+                    context.osStats(osService.stats());
+                }
+            })
+            .put(SysNodesTableInfo.Columns.HEAP, new Consumer<DiscoveryNodeContext>() {
+                @Override
+                public void accept(DiscoveryNodeContext context) {
+                    context.jvmStats(jvmService.stats());
+                }
+            })
+            .put(SysNodesTableInfo.Columns.VERSION, new Consumer<DiscoveryNodeContext>() {
+                @Override
+                public void accept(DiscoveryNodeContext context) {
+                    context.version(Version.CURRENT);
+                    context.build(Build.CURRENT);
+                }
+            })
+            .put(SysNodesTableInfo.Columns.THREAD_POOLS, new Consumer<DiscoveryNodeContext>() {
+                @Override
+                public void accept(DiscoveryNodeContext context) {
+                    ThreadPools threadPools = new ThreadPools();
```

Comment:
I think ThreadPools could have a static factory method which takes the `ThreadPool` argument..